core/src/main/java/com/datastrato/gravitino/meta/MetalakeManager.java

```diff
@@ -238,4 +242,15 @@ private BaseMetalake.Builder updateEntity(
 
     return builder.withProperties(newProps);
   }
+
+  private BaseMetalake removeIdFromProperties(BaseMetalake metalake) {
```

Comment:
This unnecessarily build the metalake again, can you please optimize this logic.